### PR TITLE
introduce 'requires OSFAMILY'

### LINKED ISSUE #2697: RFC: `@requires OSFAMILY`
Currently, `@requires OS` is checking against `PHP_OS` via regex
(`@requires OS Lin` -> `/Lin/i`).

On PHP It's tricky, as `PHP_OS` could be `WIN32`, `WINNT`, `Windows`. It would be nice to add possibility to check against `PHP_OS_FAMILY` (either by dedicated annotation or by extending what `@requires OS` checks)
WDYT?

### PR DESCRIPTION
**Pull request by keradus**
Closes #2697

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
         "phpunit/phpunit-mock-objects": "^4.0",
         "sebastian/comparator": "^2.0",
         "sebastian/diff": "^1.4.3 || ^2.0",
-        "sebastian/environment": "^3.0.2",
+        "sebastian/environment": "^3.1@dev",
```
Comment (keradus) [inline composer.json]: Could 3.1 be deployed ?
alternatively, I could use php polyfilled constant from `symfony/polyfill`.

**Comment by sebastianbergmann** [comment]
Yes, I'll release 3.1 shortly. I wanted to wait a while until the constant's value for Apple had settled.

File: src/Util/Test.php [hunk 2]
```diff
@@ -190,10 +191,14 @@ public static function getRequirements($className, $methodName)
...
-            $requires['OS'] = \sprintf(
-                '/%s/i',
-                \addcslashes($matches['value'][$count - 1], '/')
-            );
+            foreach (\range(0, $count - 1) as $i) {
+                $requires[$matches['name'][$i]] = $matches['name'][$i] === 'OSFAMILY'
+                    ? $matches['value'][$i]
+                    : \sprintf(
+                        '/%s/i',
+                        \addcslashes($matches['value'][$i], '/')
```
Comment (keradus) [inline src/Util/Test.php]: logic when to use strictly provided value and when to decorate it for regex matching can be moved direct to place, when we compare the `'OS'` value. I'm for that movement, but with that, we would need to update all related tests (to not contain `/Linux/i` but just `Linux`). WDYT @sebastianbergmann ?

**Comment by sebastianbergmann** [comment]
Makes sense to me.

File: composer.json (hunk 1 above)
Comment (sebastianbergmann) [inline composer.json]: 3.1.0 has been released

File: src/Util/Test.php (hunk 2 above)
Comment (sebastianbergmann) [inline src/Util/Test.php]: Makes sense to me.

Comment (keradus) [inline src/Util/Test.php]: done

File: composer.json (hunk 1 above)
Comment (keradus) [inline composer.json]: done

**Pull request merged**